### PR TITLE
Change 'Your account is not ready yet' page to an email address focused one

### PR DESCRIPTION
**Pull request by mnacos**
## Context

Sometimes newly invited provider users get stuck because they have previously signed up to DSI (e.g. for Publish) with a different email address and they end up using this other email address when signing into Apply. Our system does not recognise this other email address and they are shown the 'Your account is not ready' page. Unfortunately, this page is not very helpful. With more information, provider users may be able to resolve the issue themselves, instead of contacting support.

## Changes proposed in this pull request

Change the message of the page to hint the problem is the email address. The 'Your account is not ready yet' message dates back to a time when inviting users was a multi-step process. It is now fully automated/single-step. 

Before:

image

After:

image

## Guidance to review

You can log in here: https://apply-for-te-improve-yo-p1s3dt.herokuapp.com/provider/sign-in

If you use random `uid` and `email` values you'll be able to see these changes.

Is the copy ok?

## Link to Trello card

No Trello card

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/provider_interface/account_creation_in_progress.html.erb [hunk 1]
```diff
@@ -6,13 +6,24 @@
       Your account is not ready yet
```
Comment (TobyRet) [inline app/views/provider_interface/account_creation_in_progress.html.erb:6]: Potentially this line could end up being confusing given the proposed changes

File: app/views/provider_interface/account_creation_in_progress.html.erb [hunk 2]
```diff
@@ -6,13 +6,24 @@
...
     </h1>
     <p class="govuk-body">
-      You’re seeing this page because you’ve signed in
-      to the <%= t('service_name.apply') %> service but your account
-      is not ready to use yet.
+      You have signed in to DfE as
```
Comment (TobyRet) [inline app/views/provider_interface/account_creation_in_progress.html.erb:9]: Would adding this additional info still follow security good practice?
https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Authentication_Cheat_Sheet.md#authentication-and-error-messages

File: app/views/provider_interface/account_creation_in_progress.html.erb (hunk 1 above)
Comment (mnacos) [reply app/views/provider_interface/account_creation_in_progress.html.erb:6]: I agree. Happy to change this but let's wait for the final version of the copy first.

File: app/views/provider_interface/account_creation_in_progress.html.erb (hunk 2 above)
Comment (mnacos) [reply app/views/provider_interface/account_creation_in_progress.html.erb:9]: It's good to check these things, but I think the inclusion of an email address on this page is ok, as they have already authenticated by the time they see this. It's more of an authorisation error (the authenticated user is not associated with any accounts). To perform an enumeration attack, they'd basically need log in as every user they are trying to assess.

**Comment by mnacos** [comment]
Paul Genders has suggested the following copy:

**Your email address is not recognised**

You have signed into DfE as different@example. We can’t find an Apply for teacher training account associated with that address. 
Please sign in again, using the email address associated with your Apply account. 
If you are using the correct email address to sign in to Apply and you’re still receiving this message, please contact us at becoming@etc.

**Comment by adamsilver** [comment]
I think avoid bold and put that sentence into its own paragraph.

**Review by adamsilver — APPROVED** [review]

**Pull request merged**